docs/proposals/PROPOSAL_B.md

```diff
@@ -0,0 +1,287 @@
+# Proposal B - Extend Platform Variants and Introduce Compatibility Artifact
+
+This proposal recommends changes to image spec for more granular image selection and introduces a new image compatibility artifact type.
+Based on the artifact, cluster administrators can verify hosts or prepare the appropriate infrastructure for containers.
+
+## Modifications
+
+### Image Spec
+
+Extend the platform variants with new ARM and AMD64 micro-architecture levels.
+
+| ISA/ABI          | architecture | variant |
+|------------------|--------------|---------|
+| ...              | ...          | ...     |
+| ARM 64-bit, v9   | arm          | v9      |
+| AMD64 64-bit, v2 | amd64        | v2      |
+| AMD64 64-bit, v3 | amd64        | v3      |
+| AMD64 64-bit, v4 | amd64        | v4      |
+
+## Image Compatibility Artifact
+
+!["Image Compatibility Artifact"](img/B_artifact.png "Image Compatibility Artifact")
+
+### Definitions
+
+* image-compatibility artifact - an OCI artifact that stores the compatibility spec.
+* Compatibility Spec - a json object that represents a graph of namespaces.
```

Comment:
If this is to be a graph, any reason to not use a graph format for json (e.g., JGF) instead of making a new one? https://github.com/jsongraph/json-graph-specification. I made this over break: https://github.com/converged-computing/jsongraph-go

Reply:
Yeah I saw your examples. I don't have a very strong opinion here. I knew that you will propose JGF in your proposal, so I decided to propose something else, a little bit more simplified that may be build on JSON Schema directly. Better to have more choices for the discussion :)

Reply:
Since I will be using rust, anything which I can use as input such as JSON schema is preferred. Anything else is a lot of work.

Reply:
JFG is also using JSON Schema, but with the example I made I am using my own JSON Schema construction and have freedom to do anything with the spec.

In any case, we will make sure not to limit ourselves to any particular language.

Reply:
dependency also means maintenance burden, for example if bugs found in jsongraph-go package, we have to fix it, release it, update dependence. if dependence can be removed, good